### PR TITLE
initial commit of devon4ng template for PL

### PR DESCRIPTION
**Pull request by kpanter**

### CODE REVIEW

File: devon4ng-pl/Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+                }
+            }
+        }
+    
+        stage('Install angular & cicdgen') {
+            steps{
+             cleanWs()
+             deleteDir()
+
+        sh 'npm i -g @angular/cli @devonfw/cicdgen'
```
Comment (pdehnel) [inline devon4ng-pl/Jenkinsfile]: Please correct intendation :)

File: devon4ng-pl/Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+import com.capgemini.productionline.configuration.*
+
+Jenkins jenkinsConfiguration = new Jenkins();
+GitLab gitlabConfiguration = new GitLab(this, params.GITLAB_USER_PRIVATE_TOKEN, ProductionLineGlobals.GITLAB_BASE_URL);
+
+ 
+
+pipeline{
+
+    agent any
```
Comment (pdehnel) [inline devon4ng-pl/Jenkinsfile]: Please correct intendation :)

File: devon4ng-pl/Jenkinsfile [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+
+  tools {
+        nodejs "NodeJS LTS"
```
Comment (pdehnel) [inline devon4ng-pl/Jenkinsfile]: So this is a requirement ?
Then we need to add this to the documentation.

File: devon4ng-pl/Jenkinsfile [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+    }
+    //Jenkins Job Parameters
+    parameters { 
+        string(name: 'GITLAB_USER_PRIVATE_TOKEN', defaultValue: '', description: 'NEEDS TO BE SET!. Private Token of a Production Line Gitlab User that can be used to create repositories.') 
+        string(name: 'GITLAB_CREATE_GROUP_NAME', defaultValue: 'devon', description: 'Name of the group that will be created inside GitLab to clone the repo.') 
+        string(name: 'GITLAB_CREATE_PROJECT_NAME', defaultValue: 'devon4ng', description: 'Name of the repository inside GitLab that will be created.') 
+        string(name: 'GITLAB_CREATE_BRANCH', defaultValue: 'master', description: 'Branch') 
+        string(name: 'GITLAB_CREATE_PROJECT_DESCRIPTION', defaultValue: 'Devon for angular PL template', description: 'Description of the repository.') 
+        string(name: 'GITLAB_CLONE_URL', defaultValue: '', description: 'Private Token of a Production Line Gitlab User that can be used to create repositories.') 
+        string(name: 'USERNAME', defaultValue: '', description: 'Username of a Production Line Gitlab User that can be used to create repositories.')
```
Comment (pdehnel) [inline devon4ng-pl/Jenkinsfile]: only username?

File: devon4ng-pl/Jenkinsfile [hunk 5]
```diff
@@ -0,0 +1,182 @@
...
+                    println "Trying to create Build job"
+                    jobDsl scriptText: """
+                        pipelineJob("Angular/Devon4ng_build_template") {
+                            description('Build job for devon4ng template.')
+                            definition {
+                                cpsScm {
+                                    scm {
+                                        git {
+                                            remote {
+                                                url('http://gitlab-core:80/gitlab/${params.GITLAB_CREATE_GROUP_NAME}/${params.GITLAB_CREATE_PROJECT_NAME}.git')
```
Comment (pdehnel) [inline devon4ng-pl/Jenkinsfile]: Please use ProductionLineGlobals.GITLAB_BASE_URL here as well.

File: devon4ng-pl/Jenkinsfile [hunk 6]
```diff
@@ -0,0 +1,182 @@
...
+        }
+        
+        stage('Create cicdgen Angular template') {
+            steps{
+              script{
+                PL_NAME = sh(returnStdout: true, script: "curl -v -H 'Content-Type: application/json' -H 'Authorization: $MICROPORTAL_TOKEN' 'http://microportal:8080/api/metadata' | jq -r '.engagementname'")   
+                PL_NAME = PL_NAME.trim()   
+   
+                println("$PL_NAME")
+                   }
```
Comment (pdehnel) [inline devon4ng-pl/Jenkinsfile]: intendation

File: devon4ng-pl/Jenkinsfile [hunk 7]
```diff
@@ -0,0 +1,182 @@
...
+            }
+        }
+        
+        stage('Create cicdgen Angular template') {
+            steps{
+              script{
+                PL_NAME = sh(returnStdout: true, script: "curl -v -H 'Content-Type: application/json' -H 'Authorization: $MICROPORTAL_TOKEN' 'http://microportal:8080/api/metadata' | jq -r '.engagementname'")   
+                PL_NAME = PL_NAME.trim()   
+   
+                println("$PL_NAME")
```
Comment (pdehnel) [inline devon4ng-pl/Jenkinsfile]: can be removed?

File: devon4ng-pl/Jenkinsfile [hunk 8]
```diff
@@ -0,0 +1,182 @@
...
+                   }
+                sh "cd devon4ng && cicdgen generate devon4ng --docker --plurl \""+PL_NAME+".dev.s2-eu.capgemini.com\" --groupid com.devonfw"                                           
+            }
+        }
+        stage('Pushing template to Gitlab') {
+            steps{
+                
+               sh 'cd devon4ng && git remote add origin http://${USERNAME}@gitlab-core:80/gitlab/${GITLAB_CREATE_GROUP_NAME}/${GITLAB_CREATE_PROJECT_NAME}.git && git push --set-upstream origin master'
+                }
```
Comment (pdehnel) [inline devon4ng-pl/Jenkinsfile]: intendation

File: devon4ng-pl/Jenkinsfile (hunk 4 above)
Comment (kpanter) [reply devon4ng-pl/Jenkinsfile]: Yes

File: devon4ng-pl/Jenkinsfile (hunk 3 above)
Comment (kpanter) [reply devon4ng-pl/Jenkinsfile]: sure, will be added

File: devon4ng-pl/Jenkinsfile [hunk 9]
```diff
@@ -137,45 +127,37 @@ pipeline{
...
-                PL_NAME = sh(returnStdout: true, script: "curl -v -H 'Content-Type: application/json' -H 'Authorization: $MICROPORTAL_TOKEN' 'http://microportal:8080/api/metadata' | jq -r '.engagementname'")   
-                PL_NAME = PL_NAME.trim()   
-   
-                println("$PL_NAME")
-                   }
-                sh "cd devon4ng && cicdgen generate devon4ng --docker --plurl \""+PL_NAME+".dev.s2-eu.capgemini.com\" --groupid com.devonfw"                                           
+                script{
+                    PL_NAME = sh(returnStdout: true, script: "curl -v -H 'Content-Type: application/json' -H 'Authorization: $MICROPORTAL_TOKEN' 'http://microportal:8080/api/metadata' | jq -r '.engagementname'")   
+                    PL_NAME = PL_NAME.trim()   
+                      }
```
Comment (pdehnel) [inline devon4ng-pl/Jenkinsfile]: intendation

**Comment by kpanter** [comment]
Yes.

File: devon4ng-pl/Jenkinsfile (hunk 5 above)
Comment (pdehnel) [reply devon4ng-pl/Jenkinsfile]: @kpanter please use this^^

**Pull request merged**